src/Language/Python/IR/Checker/ArgumentList.hs

```diff
@@ -0,0 +1,178 @@
+  -> IR.KeywordItem name expr ann
+  -> SyntaxChecker ann (Safe.KeywordItem checkedName checkedExpr 'NotAssignable dctxt ann)
+checkKeywordItem cfg _checkName _checkExpr (IR.KeywordItem l r ann) =
+  Safe.KeywordItem <$>
+  traverseOf (_Wrapped.traverse) (_checkName cfg) l <*>
+  traverseOf
+    (_Wrapped.traverse)
+    (_checkExpr $ cfg & atomType .~ SNotAssignable)
+    r <*>
+  pure ann
+
+checkKeywordsArguments
+  :: ExprConfig as dctxt
+  -> ( forall as' dctxt'
+     . ExprConfig as' dctxt'
+    -> name ann
+    -> SyntaxChecker ann (checkedName ann))
+  -> ( forall as' dctxt'
+     . ExprConfig as' dctxt'
+    -> expr ann
+    -> SyntaxChecker ann (checkedExpr as' dctxt' ann))
+  -> IR.KeywordsArguments name expr ann
+  -> SyntaxChecker ann (Safe.KeywordsArguments checkedName checkedExpr 'NotAssignable dctxt ann)
+checkKeywordsArguments cfg _checkName _checkExpr (IR.KeywordsArguments h t ann) =
+  Safe.KeywordsArguments <$>
+  (case h of
+    InL a -> InL <$> checkKeywordItem cfg _checkName _checkExpr a
+    InR a ->
+      InR <$>
+      traverseOf
+        (_Wrapped.traverse)
+        (_checkExpr $ cfg & atomType .~ SNotAssignable)
+        a) <*>
+  traverseOf
+    (_Wrapped.traverse._Wrapped.traverse)
+    (\case
+        InL a -> InL <$> checkKeywordItem cfg _checkName _checkExpr a
+        InR a ->
+          InR <$>
+          traverseOf
+            (_Wrapped.traverse)
+            (_checkExpr $ cfg & atomType .~ SNotAssignable)
+            a)
+    t <*>
+  pure ann
+
+checkPositionalArguments
+  :: ExprConfig as dctxt
+  -> ( forall as' dctxt'
+     . ExprConfig as' dctxt'
+    -> expr ann
+    -> SyntaxChecker ann (checkedExpr as' dctxt' ann))
+  -> IR.PositionalArguments expr ann
+  -> SyntaxChecker ann (Safe.PositionalArguments checkedExpr 'NotAssignable dctxt ann)
+checkPositionalArguments cfg _checkExpr (IR.PositionalArguments h t ann) =
+  Safe.PositionalArguments <$>
+  traverseOf
+    (_Wrapped.traverse)
+    (_checkExpr $ cfg & atomType .~ SNotAssignable)
+    h <*>
+  traverseOf
+    (_Wrapped.traverse._Wrapped.traverse._Wrapped.traverse)
+    (_checkExpr $ cfg & atomType .~ SNotAssignable)
+    t <*>
+  pure ann
+
+checkStarredAndKeywords
+  :: ExprConfig as dctxt
+  -> ( forall as' dctxt'
+     . ExprConfig as' dctxt'
+    -> name ann
+    -> SyntaxChecker ann (checkedName ann))
+  -> ( forall as' dctxt'
+     . ExprConfig as' dctxt'
+    -> expr ann
+    -> SyntaxChecker ann (checkedExpr as' dctxt' ann))
+  -> IR.StarredAndKeywords name expr ann
+  -> SyntaxChecker ann (Safe.StarredAndKeywords checkedName checkedExpr 'NotAssignable dctxt ann)
+checkStarredAndKeywords cfg _checkName _checkExpr (IR.StarredAndKeywords h t ann) =
+  Safe.StarredAndKeywords <$>
+  (case h of
+     InL a ->
+       InL <$>
+       traverseOf
+         (_Wrapped.traverse)
+         (_checkExpr $ cfg & atomType .~ SNotAssignable)
+         a
+     InR a -> InR <$> checkKeywordItem cfg _checkName _checkExpr a) <*>
```

Comment:
`instance Bifunctor …`

Reply:
If anything, I think it would be Bitraversable, but `Sum` isn't Bitraversable.

I want this function:
`??? :: (f a -> m (f a)) -> (g a -> m (g a)) -> Sum f g a -> m (Sum f g a)`